src/services/completions.ts

```diff
@@ -2084,6 +2084,10 @@ export function getCompletionEntriesFromSymbols(
             continue;
         }
 
+        if (!isTypeOnlyLocation && isInJSFile(sourceFile) && !symbolHasNoTypeOnlyDeclaration(symbol)) {

```

Comment:
Double-negation coming from the name of this function and `!` makes it harder to reason about for me. I'm also a little bit wary of saying it's currently correct - I'm simply unsure. Shouldn't we check if a symbol **has a value declaration** instead of checking if it has no type-only declarations? "Mixed" symbols are always somewhat confusing and I simply avoid them in my own code so I'm not sure what are the exact rules around them. Plus I'm not sure how that combines with `.js` files as that's also something I don't use too often.

Reply:
Hmm yeah added some comments & improved the name.